metals/src/main/scala/scala/meta/internal/metals/StandaloneSymbolSearch.scala

```diff
@@ -95,36 +95,87 @@ object StandaloneSymbolSearch {
       buffers: Buffers,
       sources: Seq[Path],
       classpath: Seq[Path],
-      isExcludedPackage: String => Boolean
+      isExcludedPackage: String => Boolean,
+      userConfig: () => UserConfiguration
   ): StandaloneSymbolSearch = {
+    val (sourcesWithExtras, classpathWithExtras) =
+      addScalaAndJava(
+        sources.map(AbsolutePath(_)),
+        classpath.map(AbsolutePath(_)),
+        userConfig().javaHome
+      )
+
     new StandaloneSymbolSearch(
       workspace,
-      classpath.map(path => AbsolutePath(path)),
-      sources.map(path => AbsolutePath(path)),
+      classpathWithExtras,
+      sourcesWithExtras,
       buffers,
       isExcludedPackage
     )
   }
   def apply(
       workspace: AbsolutePath,
       buffers: Buffers,
-      isExcludedPackage: String => Boolean
+      isExcludedPackage: String => Boolean,
+      userConfig: () => UserConfiguration
   ): StandaloneSymbolSearch = {
-    val scalaVersion = BuildInfo.scala212
-
-    val jars = Embedded
-      .downloadScalaSources(scalaVersion)
-      .map(path => AbsolutePath(path))
-
-    val (sources, classpath) =
-      jars.partition(_.toString.endsWith("-sources.jar"))
+    val (sourcesWithExtras, classpathWithExtras) =
+      addScalaAndJava(Nil, Nil, userConfig().javaHome)
 
     new StandaloneSymbolSearch(
       workspace,
-      classpath,
-      sources,
+      classpathWithExtras,
+      sourcesWithExtras,
       buffers,
       isExcludedPackage
     )
   }
+
+  /**
+   * When creating the standalone check to see if the sources have Scala. If
+   * not, we add the sources and the classpath. We also add in the JDK sources.
+   *
+   * @param sources the original sources that may be included.
+   * @param classpath the current classpath.
+   * @param javaHome possible JavaHome to get the JDK sources from.
+   * @return (scalaSources, scalaClasspath)
+   */
+  private def addScalaAndJava(
+      sources: Seq[AbsolutePath],
+      classpath: Seq[AbsolutePath],
+      javaHome: Option[String]
+  ): (Seq[AbsolutePath], Seq[AbsolutePath]) = {
+    val missingScala: Boolean =
+      sources.filter(_.toString.contains("scala-library")).isEmpty
+
+    (missingScala, JdkSources(javaHome)) match {
+      case (true, None) => getScala(BuildInfo.scalaCompilerVersion)
```

Comment:
That will be `BuildInfo.scala212` in reality since Metals is compiled with 2.12

Reply:
Yea, I knew it'd be the same, but I figured I'd add in `scalaCompilerVersion` to not have to change it in the future if we change to 2.13 for example. I can change it back to `BuildInfo.scala212` if  you want?

Reply:
We can also use the one from mtags. There is already `scala.meta.internal.mtags.BuildInfo` with `scalaCompilerVersion` which will correspond to the metals version.

It's also fine to add an additional field, either way is fine.

Reply:
Alright, if you're fine with it, I'll leave this as is and still use `scalaCompilerVersion` here since again _hopefully_ soon if we bump to 2.13, we can then default stuff like this to that and not have to go around changing it.

> We can also use the one from mtags.

But we don't have access to the mtags BuildInfo here right?

Reply:
We do have access, since Metals itself is using the 2.12 jar to access all the classes defined in the mtags module.

Reply:
Huh, I guess I've just never done that before and reached into the BuildInfo of another module. Anyways, I changed this to do that.  It now grabs the `scalaCompilerVersion` from mtags.